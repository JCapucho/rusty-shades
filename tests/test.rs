use rusty_shades::build_naga_ir;
use std::fs::read_to_string;

#[test]
fn insta_test() {
    insta::glob!("*.rsh", |file| {
        let code = read_to_string(file).unwrap();
        let res = build_naga_ir(&code);
        insta::assert_ron_snapshot!(res);
    });
}
