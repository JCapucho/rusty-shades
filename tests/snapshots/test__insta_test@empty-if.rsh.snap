---
source: tests/test.rs
expression: res
input_file: tests/empty-if.rsh
---
Ok(Module(
  header: Header(
    version: (1, 0, 0),
    generator: 1920299892,
  ),
  types: [],
  constants: [
    Constant(
      name: None,
      specialization: None,
      inner: Scalar(
        width: 4,
        value: Bool(true),
      ),
    ),
  ],
  global_variables: [],
  functions: [
    Function(
      name: Some("test"),
      arguments: [],
      return_type: None,
      global_usage: [],
      local_variables: [],
      expressions: [],
      body: [],
    ),
  ],
  entry_points: {
    (Vertex, "vertex_main"): EntryPoint(
      early_depth_test: None,
      workgroup_size: (0, 0, 0),
      function: Function(
        name: None,
        arguments: [],
        return_type: None,
        global_usage: [],
        local_variables: [],
        expressions: [
          Constant(1),
        ],
        body: [
          If(
            condition: 1,
            accept: [
              Kill,
            ],
            reject: [
              Kill,
            ],
          ),
          Return(
            value: None,
          ),
        ],
      ),
    ),
  },
))
